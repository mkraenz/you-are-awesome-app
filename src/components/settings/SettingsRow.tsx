import React, { FC, ReactNode } from "react";
import { List } from "react-native-paper";

const SettingsRow: FC<{
    onPress: () => void;
    title: string;
    rightComponent?: () => ReactNode;
    disabled?: boolean;
}> = ({ onPress, title, rightComponent, disabled }) => (
    <List.Item
        onPress={onPress}
        key={title}
        title={title}
        right={rightComponent}
        disabled={disabled}
    ></List.Item>
);

export default SettingsRow;
