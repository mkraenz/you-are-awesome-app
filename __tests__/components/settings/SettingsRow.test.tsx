import React from "react";
import "react-native";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import SettingsRow from "../../../src/components/settings/SettingsRow";

it("renders correctly", () => {
    const tree = renderer
        .create(<SettingsRow onPress={() => undefined} title={"MyTitle"} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
