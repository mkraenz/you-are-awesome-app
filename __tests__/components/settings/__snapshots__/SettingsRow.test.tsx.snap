// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      false,
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginVertical": 6,
            "paddingLeft": 8,
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
              },
              undefined,
            ],
          ]
        }
      >
        MyTitle
      </Text>
    </View>
  </View>
</View>
`;
